Task 2 completed

SashiFazya opened the pull request.

SashiFazya committed:
Task 2 completed

nikolay-977 commented on src/main/java/LoginData.java:
```diff
@@ -0,0 +1,31 @@
+public class LoginData {
```
Можно улучшить: Хорошая практика - размещать классы в пакетах, объединяя их по функциональности (классы с запросами в src.main.java.api.client, классы с описанием модели данных в src.main.java.api.model, а в вспомогательные классы в src.main.java.api.util) либо по сущностям (src.main.java.api.courier, src.main.java.api.order)

nikolay-977 commented on src/main/java/OrderMethods.java:
```diff
@@ -0,0 +1,50 @@
+    private static final String GET_AUTH_USER_ORDERS = "/api/orders";
+    private UserMethods userMethods = new UserMethods();
+
+    public ValidatableResponse createOrderUnauthorized(Order order) {
```
Необходимо исправить: При реализации шагов используй аннотацию `@Step("Создание заказа")`. Без аннотации `@Step` шаг не отобразится в Allure отчете. В скобках указывай описание шага. Это сделает отчет Allure информативнее, что позволит сократить время локализации дефекта.

nikolay-977 commented on src/test/java/TestGetUserOrders.java:
```diff
@@ -0,0 +1,55 @@
+                .assertThat().statusCode(SC_OK)
+                .assertThat().body("orders", notNullValue(),
+                        "success", equalTo(true))
+                .log().all();
```
Необходимо исправить: .log().all() следует вызывать на уровне спецификации либо методов классов OrderMethods и UserMethods. Это позволит избежать повторяющегося кода и сократит код автотеста.

nikolay-977 commented on src/test/java/TestUserUpdate.java:
```diff
@@ -0,0 +1,98 @@
+                .log().all();
+
+        //для удаления обновим данные юзера
+        user.setName(newName);
```
Необходимо исправить: Проверки должны быть в конце тестовых методов. Между проверками и после проверок не должно быть дополнительных действий. Иначе в случае фейла первой проверки, последующие действия(шаги и методы) не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта.

SashiFazya replied:
Не понимаю, как это исправить... Новое значение хранится именно локально в тесте, вынести это на уровень выше или перед ассертом не получится

nikolay-977 replied:
1. Для удаления пользователя достаточно передать accessToken, объект user передавать не нужно, метод удаления пользователя следует реализовать так:

```
    @Step("Удалить пользователя {user.email}")
    public ValidatableResponse deleteUser(String accessToken) {
        return given()
                .spec(getSpec())
                .header("Authorization", accessToken).log().all()
                .delete(DELETE_USER_URL)
                .then().log().all();
    }
```
2. Метод для вызова метода удаления пользователя реализовать так:

```
    @After
    public void cleanUp() {
        if (accessToken != null)
            deleteUser(accessToken).statusCode(SC_ACCEPTED);
    }
```

2. В классе TestUserUpdate (и в классах где требуется удаление) нужно объявить переменную  `private accessToken;` и перед проверками вызвать метод для получения токена и сохранять значение в переменную accessToken:

```
        ValidatableResponse  updateAuthorizedResponse = updateAuthorizedUserData(user, json)
        accessToken = userMethods.getAccessToken(user);        
        updateAuthorizedResponse.assertThat().statusCode(SC_OK)
                .assertThat().body("success", equalTo(true),
                        "user.email", equalTo(originalEmail),
                        "user.name", equalTo(newName));
```

SashiFazya replied:
Так гораздо красивше) спасибо!

SashiFazya committed:
Разложила классы по пакетам.
Добавила аннотацию Step к методам client классов.
Добавила модификаторы доступа к инстанс-переменным.
log().all() перенесла в методы.
Не исправила вызов метода после assert - не понимаю, как решить проблему с удалением пользователя после изменения данных без этого вызова.

SashiFazya committed:
Изменила метод deleteUser
